### ISSUE TITLE
Sign NuGet packages prior to publish

### ISSUE THREAD

**Issue by AaronRobinsonMSFT** [dotnet/coreclr#26051]
In PR https://github.com/dotnet/coreclr/pull/26020 sign checking has been deferred. This is being deferred because the coreclr doesn't presently sign `nupkg` files. Once that is done, sign validation can be re-enabled.

cc @dotnet/coreclr-infra

**Comment by AaronRobinsonMSFT** [comment]
Based on guidance from the Arcade team, signing NuGet packages that aren't being shipped isn't needed. Closing this as support for validating contents of an unsigned package is now supported. See https://github.com/dotnet/arcade/pull/3611.
